lightningd/invoice.c

```diff
@@ -81,7 +81,7 @@ static void tell_waiter(struct command *cmd, const struct invoice *paid)
 	json_object_start(response, NULL);
 	json_add_string(response, "label", paid->label);
 	json_add_hex(response, "rhash", &paid->rhash, sizeof(paid->rhash));
-	json_add_u64(response, "msatoshi", paid->msatoshi);
+	json_add_u64(response, "msatoshi", *paid->msatoshi);
 	json_add_bool(response, "complete", paid->state == PAID);
 	if (paid->state == PAID)
 		json_add_u64(response, "pay_index", paid->pay_index);
```

Comment:
This is, in fact, always true here.  (in tell_waiter).  Probably should just assert paid->state == PAID at the top.

Maybe we should share this json-printing-invoice routine somewhere?

Also, looks like we screwed up sometime, and this routine is called in two loops; that should be fixed too, while you're poking around :)

Reply:
Probably indeed better to factor out json-printing an invoice into common/json.c instead.

Re: two loops in `resolve_invoice`, hahahaha, looks like a rebase of waitanyinvoice and/or waitinvoice things I did a short while ago got borked, will make a separate pull request for that.